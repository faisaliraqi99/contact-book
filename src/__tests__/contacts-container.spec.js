import React from 'react'
import { shallow, mount } from 'enzyme'
import { createStore } from 'redux'
import rootReducer from '../reducers'
import { BrowserRouter as Router } from 'react-router-dom'

import ContactsContainer, { ContactsContainer as OriginalContactsContainer } from '../container/contacts-container'

const initialData = {
  contacts: [
    {
      id: 1,
      name: 'Somename',
      lastname: 'Lastname',
      number: '550121212',
      countrycode: '+996',
      email: 'undefined@mail.ru',
      address: 'Someaddr'
    },
    {
      id: 2,
      name: 'Somename',
      lastname: 'Lastname',
      number: '550121212',
      countrycode: '+996',
      email: 'undefined@mail.ru',
      address: 'Someaddr'
    }
  ]
}

const store = createStore(rootReducer);

describe('Contacts-container render',() => {
  
  it('ContactsContainer render .contact-list', () => {
    const ContactsContainerMount = mount(<ContactsContainer store={store} />);
    expect(ContactsContainerMount.find('.contact-list')).toHaveLength(1);
  });
  
  it('ContactsContainer snapshot test', () => {
    const ContactsContainerShallow = shallow(<ContactsContainer store={store} /> );
    expect(ContactsContainerShallow).toMatchSnapshot();
  });

});


describe('Contacts-container selectItem test', () => {

  const wrapper = mount(
    <Router>
      <OriginalContactsContainer store={store} state={initialData} />
    </Router>);

  beforeEach(() => {
    wrapper.find('li').first().simulate('click');
  });

  it('Contacts-container selectItem work correct', () => {
    const ReduxState = wrapper.find('ContactsContainer').props().store.getState();
    expect(ReduxState.state.selectedContact).toEqual(initialData.contacts[0]);
  });

});