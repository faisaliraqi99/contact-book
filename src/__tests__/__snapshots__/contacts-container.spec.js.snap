// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts-container render ContactsContainer snapshot test 1`] = `
<ContextProvider
  value={null}
>
  <ContactsContainer
    actionEditContact={[Function]}
    addSelectedContact={[Function]}
    createContact={[Function]}
    createContactSuccess={[Function]}
    deleteContact={[Function]}
    fetchAllContacts={[Function]}
    fetchContact={[Function]}
    state={
      Object {
        "contacts": Array [],
        "isLoading": null,
        "selectedContact": Object {
          "address": "Select Contact",
          "email": "Select Contact",
          "lastname": "Select Contact",
          "name": "Select Contact",
          "number": "Select Contact",
        },
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
